File: android/src/main/java/com/getcapacitor/community/classes/results/LicenseExpirationDateResult.java
package com.getcapacitor.community.classes.results;

import com.getcapacitor.JSObject;
import com.getcapacitor.community.interfaces.Result;

import java.util.Date;

public class LicenseExpirationDateResult implements Result {

    private final Date licenseExpirationDate;

    public LicenseExpirationDateResult(Date licenseExpirationDate) {
        this.licenseExpirationDate = licenseExpirationDate;
    }

    @Override
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        if (licenseExpirationDate != null) result.put("licenseExpirationDate", licenseExpirationDate.toString());

        return result;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Results/LicenseExpirationDateResult.swift
import Foundation
import Capacitor

@objc public class LicenseExpirationDateResult: NSObject, Result {
    let licenseExpirationDate: Date?

    init(_ licenseExpirationDate: Date?) {
        self.licenseExpirationDate = licenseExpirationDate
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()

        if let licenseExpirationDate = licenseExpirationDate {
            result["licenseExpirationDate"] = licenseExpirationDate.timeIntervalSince1970 * 1000
        }

        return result as AnyObject
    }
}
